src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs

```diff
@@ -867,6 +867,92 @@ internal sealed partial class LazyBinaryOperatorExpression : BaseBinaryOperatorE
         protected override IOperation RightOperandImpl => _lazyRightOperand.Value;
     }
 
+    /// <summary>
+    /// Represents an operation with two operands that produces a result with the same type as at least one of the operands.
+    /// </summary>
+    internal abstract partial class BaseTupleBinaryOperatorExpression : Operation, ITupleBinaryOperation
+    {
+        public BaseTupleBinaryOperatorExpression(BinaryOperatorKind operatorKind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit)
+            : base(OperationKind.BinaryOperator, semanticModel, syntax, type, constantValue, isImplicit)
+        {
+            OperatorKind = operatorKind;
+        }
+        /// <summary>
+        /// Kind of binary operation.
+        /// </summary>
+        public BinaryOperatorKind OperatorKind { get; }
+        /// <summary>
+        /// Left operand.
+        /// </summary>
+        public IOperation LeftOperand => Operation.SetParentOperation(LeftOperandImpl, this);
+        /// <summary>
+        /// Right operand.
+        /// </summary>
+        public IOperation RightOperand => Operation.SetParentOperation(RightOperandImpl, this);
+
+        protected abstract IOperation LeftOperandImpl { get; }
+        protected abstract IOperation RightOperandImpl { get; }
+
+        public override IEnumerable<IOperation> Children
+        {
+            get
+            {
+                if (LeftOperand != null)
+                {
+                    yield return LeftOperand;
+                }
+                if (RightOperand != null)
+                {
+                    yield return RightOperand;
+                }
+            }
+        }
+
+        public override void Accept(OperationVisitor visitor)
+        {
+            visitor.VisitTupleBinaryOperator(this);
+        }
+
+        public override TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
+        {
+            return visitor.VisitTupleBinaryOperator(this, argument);
+        }
+    }
+    /// <summary>
+    /// Represents an operation with two operands that produces a result with the same type as at least one of the operands.
+    /// </summary>
+    internal sealed partial class TupleBinaryOperatorExpression : BaseTupleBinaryOperatorExpression, ITupleBinaryOperation
```

Comment:
`ITupleBinaryOperation` is unnecessary here and in other implementation.

Reply:
I have a bug on me to make this file generated, but it's relatively low on the priority list at the moment: [https://github.com/dotnet/roslyn/issues/20204](https://github.com/dotnet/roslyn/issues/20204)

---
In reply to: [173868128](https://github.com/dotnet/roslyn/pull/25374#discussion_r173868128) [](ancestors = 173868128,173851840)